tiledb/sm/subarray/subarray.cc

```diff
@@ -77,7 +81,11 @@ Subarray::Subarray(
     const Layout layout,
     Stats* const parent_stats,
     const bool coalesce_ranges)
-    : stats_(parent_stats->create_child("Subarray"))
+    : sp_stats_(
```

Comment:
I believe it will only wind up with one stats_.
Prior core use of Subarray is presumably coming from somewhere that can provide a valid stats.
tiledb_subarray_alloc() does not have a valid stats to provide and passes a nullptr.
What is implemented here is what I came up with after merging changes that implemented the parent_stats changes that seems to at least mechanically satisfy the stats_ requirement.
Should, perhaps instead, a tiledb_subarray_t have a stats entity member added, and tiledb_subarray_alloc() thus allocate a stats entity to be passed to the Subarray ctor, and tiledb_subarray_free() subsequently free that entity?
Should tiledb_subarray_alloc() be changed to require a stats entity (affecting also at least tiledb_query_get_subarray() and some cppapi Subarray construction)?

Reply:
Ok I took a quick look and I think stats should only be used when operations from the query are happening (one more point to reinforce that we shouldn't expose the size estimation apis at the subaray level). When a subarray gets assigned to a query, we pass in the parent stats from the query to the subarray. This will remove the need for sp_stats_.

Reply:
There is a need for some sort of update parent/stats update when/if an external instance is placed into a Query.

Reply:
However, what is your objection to sp_stats_, or at least something similar?
I currently see no particular benefit to removing it given the following downsides, 
and therefore benefit to keeping it, 
A) to avoid having to modify/touch at least 19 places in the code, thus 19 chances to make some mistake
and
B) avoiding adding considerable visual noise in the places that would be touched, 
each requiring an if() or ternary-op when all of those places can be left as is by having something of the sp_stats_ approach in place...

Reply:
^^ @KiterLuc

Reply:
An idea, when it's null, call storage_manager_->stats()->create_child("UnassignedSubarray"), when the subarray gets assigned to a query, change stats_ to link to the parent by using query->stats()->create_child("Subarray"). Thoughts?

Reply:
Then we can totally remove sp_stats. It's just confusing to have two stats members on an object... It confused Joe, and me as well :)

Reply:
Any updates on this one?

Reply:
sp_stats was removed, (apparently I failed to commit earlier comment regarding)
I did not create any sort of ticket, as the 'storage_manager_->stats()->create_child(...) seemed acceptable, both to you and me.